rust/src/bin_resource_pack.rs

```diff
@@ -0,0 +1,196 @@
+//! This file contains the code for the resource-pack executable.
+//!
+//! A resource pack is a json file that identifies the resources in a particular game version/mod.
+//!
+//!
+//! # Create a resource pack:
+//!
+//! Example:
+//! ```
+//! resource-pack create --name "My resource pack" --pretty --file-size --hash md5 --datadir /path/to/game/dir --property vanilla_version ENGLISH --output pack.json
+//! ```
+//!
+
+mod res;
+mod slf;
+
+use res::{ResourcePackBuilder, ResourcePropertiesExt};
+
+use clap::{crate_version, App, Arg, ArgMatches, SubCommand};
+use serde_json;
+use std::fmt::Debug;
+use std::fs;
+use std::process;
+
+/// Entry point of the resource-pack executable.
+fn main() {
+    let cmd_create = SubCommand::with_name("create")
+        .about("Creates a resource pack.")
+        .version("1.0")
+        .arg(
+            Arg::with_name("file-size")
+                .help("Adds the file size to the resource properties")
+                .long("file-size"),
+        )
+        .arg(
+            Arg::with_name("pretty")
+                .help("Outputs with the pretty formatter")
+                .long("pretty"),
+        )
+        .arg(
+            Arg::with_name("archive")
+                .help("Adds the archive contents as resources")
+                .long("archive")
+                .value_name("TYPE")
+                .possible_values(&["slf"])
+                .takes_value(true)
+                .multiple(true),
+        )
+        .arg(
+            Arg::with_name("datadir")
```

Comment:
Is this really the data dir? Or is a directory containing the data dir? We had that mixup in the launcher once and it was not really clear to users.

Reply:
It is the same thing that is being stored in the options and that ja2 receives as a command line argument.
Personally I find it confusing since it is actually the game dir, where a "Data" directory exists.
But it's what you use right now so I just followed along.

Reply:
Yes that still is the issue I explained above. I'm fine with leaving it as it is right now and rename it everywhere at some point.

Reply:
I'll change the names in my code to gamedir.